romancal/regtest/test_wfi_pipeline.py

```diff
@@ -95,6 +95,22 @@ def test_level2_image_processing_pipeline(rtdata, ignore_asdf_paths):
                       passfail(model.meta.cal_step.saturation == 'COMPLETE'))
     assert model.meta.cal_step.saturation == 'COMPLETE'
 
+    # SOC-587 tests for WFI mode
+    if model.meta.exposure.type == 'WFI_IMAGE':
+        # check if assign_wcs step is complete
+        pipeline.log.info('SOC-587 MSG: Status of the step:             assign_wcs    ' +
+                      str(model.meta.cal_step.assign_wcs))
+        pipeline.log.info('SOC-587 MSG: Testing completion of WCS assignment in'
+                        'Level 2 image output.......' +
+                        passfail(model.meta.cal_step.assign_wcs == 'COMPLETE'))
+        assert model.meta.cal_step.assign_wcs == 'COMPLETE'
+        # check if WCS exists
+        pipeline.log.info('SOC-587 MSG: Testing that a WCS object exists    ')
+        pipeline.log.info('SOC-587 MSG: Testing that WCS exists in'
+                        'Level 2 image output.......' +
+                        passfail(model.meta.wcs is not None))
+        assert model.meta.wcs is not None
+
```

Comment:
I'm not sure what the best way is to show there's a distortion transformation in the WCS. May be check that a coordinate frame V2V3 exists in the WCS pipeline as in 
`assert "v2v3" in model.meta.wcs.available_frames`

Reply:
In addition to that check it should be straightforward to generate coordinate position before and after the distortion correction, verify the value before and after and that they are different, c.f. test_distortion in gwcs.

Reply:
> I'm not sure what the best way is to show there's a distortion transformation in the WCS. May be check that a coordinate frame V2V3 exists in the WCS pipeline as in `assert "v2v3" in model.meta.wcs.available_frames`

I added a test to assert whether or not `v2v3` is available in the `available_frames` list.

Reply:
> Also the requirement is "SOC-587: In Wide Field Imaging mode, the DMS shall generate Level 2 science data products that incorporate information needed to correct for pixel geometric distortion." so the log messages should reference that the geometric distortion correction information is available.

Added check for coords before and after distortion correction.